common/llvm.saw

```diff
@@ -1,28 +1,51 @@
 // LLVM utility definitions
 
-// Returns a newly-allocated pointer of the given type that points to
-// the given value.
-let alloc_init ty v = do {
-    p <- crucible_alloc ty;
-    crucible_points_to p v;
-    return p;
+// Remove 'crucible' from common commands
+let points_to = crucible_points_to;
+let precond = crucible_precond;
+let from_cryptol = crucible_term;
+let execute = crucible_execute_func;
+let postcond = crucible_postcond;
+let returns = crucible_return;
+let struct = crucible_struct;
+let array = crucible_array;
+let llvm_verify module function_name overrides spec =
+  crucible_llvm_verify module function_name overrides true spec z3;
+
+// Declare constants used to denote whether an allocation is
+// 'READONLY' (CONST) or not (WRITE).
+let CONST = {{ False }};
+let WRITE = {{ True }};
+
+// Returns a newly-allocated pointer of the given type.
+let pointer (rw : Term) (type : LLVMType) = if (eval_bool {{ rw == WRITE }})
+                      then crucible_alloc type
+                      else crucible_alloc_readonly type;
+
+// Returns both the SetupValue and a Term for a new symbolic
+// variable.
+let variable (name : String) (type : LLVMType) = do {
+  cryptol_var <- crucible_fresh_var name type;
+  return {s=from_cryptol cryptol_var : SetupValue, t=cryptol_var : Term};
 };
 
-// Returns a fresh value of the given type along with a newly-allocated
-// pointer that points to that value.
-let ptr_to_fresh n ty = do {
-    x <- crucible_fresh_var n ty;
-    p <- alloc_init ty (crucible_term x);
-    return (x, p);
+// Returns a newly-allocated pointer of the given type that points to
+// a new symbolic variable. Also returns the SetupValue and Term for
+// the variable.
+let alloc (name : String) (rw : Term) (type : LLVMType) = do {
+  p <- pointer rw type;
```

Comment:
I wonder if this should be called `alloc_init` and the next one `alloc_value` or something. Because this one _does_ initialize it, just to a fresh variable.